variable.c

```diff
@@ -1377,6 +1383,8 @@ rb_obj_convert_to_too_complex(VALUE obj, st_table *table)
 
     VALUE *old_ivptr = NULL;
 
+    rb_ivar_foreach(obj, rb_obj_write_barrier_ivars_i, (st_data_t)obj);
```

Comment:
I don't understand why this is needed. But if I remove it, I get some GC crash trying to mark T_NONE.